dependencies/http/http.go

```diff
@@ -51,10 +54,21 @@ func (l roundTripLimiter) RoundTrip(r *http.Request) (*http.Response, error) {
 	return response, nil
 }
 
+// Check all redirects for invalid URLs.
+func checkRedirect(req *http.Request, via []*http.Request) error {
+	if len(via) >= 10 {
+		return errors.New("stopped after 10 redirects")
+	}
+
+	validator := url.PrivateIPValidator{}
```

Comment:
We need this validator to be set via the dependencies, instead of hard coding it to always validate PrivateIPs.

Reply:
Ah, yeah, that's the thing that didn't make sense to me. Thanks for the check.